### PR TITLE
make permission constant Overridable 

### PR DESCRIPTION
**Pull request by sanjaykumarcg**
Adresses/ #435.

Implements
These pull request is trying to  define permission constant overriding .
* 
* 
* 

@devonfw/cobigen

### CODE REVIEW

File: cobigen-templates/templates-oasp4j/src/main/java/com/devonfw/cobigen/templates/oasp4j/constants/Field.java [hunk 1]
```diff
@@ -43,5 +51,10 @@
     public String toString() {
         return value;
     }
+    
+    public static void main(String arg[]) {
+    	Field.NAME.setValue("hello");
+    	System.out.println("name= "+ Field.NAME.getValue() );
+    }
```
Comment (maybeec) [inline cobigen-templates/templates-oasp4j/src/main/java/com/devonfw/cobigen/templates/oasp4j/constants/Field.java]: This code as well as the getters/setters, does not seem to be useful for this PR in any case. Most probably, you have used it for testing. Please remove testing code.

File: cobigen-templates/templates-oasp4j/src/main/templates/constants/security_permissions/templates.xml [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
-    <templates>
-        <template name="permission_constants" destinationPath="src/main/java/${variables.rootPackage}/general/common/api/constants/PermissionConstants.java" templateFile="PermissionConstants.java.ftl"/>
-    </templates>
-    <increments>
-        <increment name="permission_constants" description="Permission Constants">
-            <templateRef ref="permission_constants"/>
-        </increment>
-    </increments>
-</templatesConfiguration>
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (maybeec) [inline cobigen-templates/templates-oasp4j/src/main/templates/constants/security_permissions/templates.xml]: The formatting is odd. Please edit xml and java files just in eclipse. Nowhere else.

File: cobigen-templates/templates-oasp4j/src/main/templates/constants/security_permissions/templates.xml [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+<templatesConfiguration xmlns="http://capgemini.com/devonfw/cobigen/TemplatesConfiguration" version="2.1">
+    <templates>
+        <template name="permission_constants" destinationPath="src/main/java/${variables.rootPackage}/general/common/api/constants/PermissionConstants.java" templateFile="PermissionConstants.java.ftl" mergeStrategy="tsmerge"/>
```
Comment (maybeec) [inline cobigen-templates/templates-oasp4j/src/main/templates/constants/security_permissions/templates.xml]: You cannot merge java code (destinationPath) using typescript merger.

**Comment by jdiazgon** [comment]
@maybeec Can you check if the formatting is correct now? I have checked on his laptop and it seems correct to me, but on GitHub I see some strange tabs.

**Pull request closed without merging**